app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -21,12 +26,25 @@ open class FeedViewModel : ViewModel() {
 
     init {
         refresh()
+        feedItems = feedRepository.getFeedItems()
+        isEmpty.addSource(feedRepository.getFeedItems()){items -> isEmpty.postValue(items.isEmpty())}
     }
 
     fun refresh() {
-        //todo: fix the implementation
-        isLoading.value = false
-        isEmpty.value = true
+        isLoading.value = true
+        feedRepository.refresh()
+            .subscribeOn(Schedulers.io())
+            .subscribe({ handleResponse() },{ throwable-> handleNetworkError(throwable)})
```

Comment:
Since you don't need to write to the database in `handleResponse()` anymore, you can add `observeOn(AndroidSchedulers.mainThread())` and then write just `isLoading.value = false`